packages/govuk-frontend/src/govuk/components/skip-link/skip-link.mjs

```diff
@@ -31,28 +33,41 @@ export class SkipLink extends GOVUKFrontendComponent {
     this.$module = $module
 
     // Check for linked element
-    const $linkedElement = this.getLinkedElement()
-    if (!$linkedElement) {
-      return
+    try {
+      const $linkedElement = this.getLinkedElement()
+      this.$linkedElement = $linkedElement
+    } catch (err) {
+      throw new MissingElementError(
+        `Skip link: ${
+          err instanceof Error ? err.message : 'Linked element not found'
+        }`
+      )
     }
 
-    this.$linkedElement = $linkedElement
     this.$module.addEventListener('click', () => this.focusLinkedElement())
   }
 
   /**
    * Get linked element
    *
    * @private
-   * @returns {HTMLElement | null} $linkedElement - DOM element linked to from the skip link
+   * @throws {Error} If the "href" attribute does not contain a hash
+   * @throws {TypeError} If the element with the specified ID is not found
+   * @returns {HTMLElement} $linkedElement - DOM element linked to from the skip link
    */
   getLinkedElement() {
     const linkedElementId = this.getFragmentFromUrl()
     if (!linkedElementId) {
-      return null
+      throw new Error(`$module "href" attribute does not contain a hash`)
+    }
+
+    const linkedElement = document.getElementById(linkedElementId)
+
+    if (!linkedElement) {
+      throw new TypeError(`Target selector "#${linkedElementId}" not found`)
```

Comment:
Not a blocker, but we now say "linked element" in the fallback error but "target selector" here

```suggestion
      throw new TypeError(`Linked element selector "#${linkedElementId}" not found`)
```